app/controllers/subscriptions_controller.rb

```diff
@@ -44,13 +44,35 @@ def destroy
     redirect_to subscriptions_path
   end
 
-  def toggle_active
+  def activate
+    verify_phone_number
+  end
+
+  def pause
     @subscription.toggle!(:active)
     redirect_to subscriptions_path
   end
 
   private
 
+  def start_verification
```

Comment:
These 3 functions are written well but I think they can be moved to another service class, such as subscription_verifier